### ISSUE TITLE
Openbr - Segmentation Fault

### ISSUE THREAD

**Issue by ShwetaSood** [biometrics/openbr#351]
![screenshot from 2015-03-23 01 05 25](https://cloud.githubusercontent.com/assets/6500517/6771035/c3dd0284-d0f8-11e4-8769-fdf72470b7f9.png)
I have tried installing openbr on with opencv-2.4.9
Used this link :http://openbiometrics.org/doxygen/latest/linux_gcc.html
However, I had openCV preinstalled so skipped step no 3.
running make test is giving segmentation fault for all the tests.
Please Help!

**Comment by peterpans01** [comment]
I have same issue when I run make test. It complains about the test 4,5,7,8,9. The log is below: 

LOG: 
******************************************************************************************************************
Loading FaceRecognition
Warning: qUncompress: Z_DATA_ERROR: Input data is corrupted
  SDK Path: /home/hunnguye/project/testing_3rdLibs/openbr
  File: tools/qbytearray.cpp
  Function: QByteArray qUncompress(const uchar*, int)
  Line: 602
Fatal: Transform registry does not contain object named: 
  SDK Path: /home/hunnguye/project/testing_3rdLibs/openbr
  File: /home/hunnguye/project/testing_3rdLibs/openbr/openbr/openbr_plugin.h
  Function: static T* br::Factory<T>::make(const br::File&) [with T = br::Transform]
  Line: 898
<end of output>
Test time =   1.52 sec
----------------------------------------------------------
Test Failed.

*******************************************************************************************************************

All of them complain about the same problem ( which I think about the path of input but I can confirm that the path is correct). I still try to figure it out. Please hep :)

**Comment by jklontz** [comment]
What version of Qt?

**Comment by peterpans01** [comment]
I run it in Fedora 19 with machine have both qt3, qt4 and qt5 (v 5.3.2 after checked, all modules of qt5 was installed.).

**Comment by jklontz** [comment]
I would double check that only Qt 5 libraries are being loaded. My guess is that for some reason `qUncompress` is being pulled from Qt 4.

**Comment by peterpans01** [comment]
I can confirm that "qUncompress" is from qByteArray of qt5. I used QtCreator to follow the link between source code (Ctrl + click ->.....). With the example I used, Age Estimation, it has some trouble with read the model AgeRegression.  Somehow when it try to create a plugin from file with empty string name (with debug mode: I saw the value of name = ".").

**Comment by peterpans01** [comment]
How do you create file AgeRegression? It's compressed by zlib not come from Qt?

**Comment by jklontz** [comment]
Following the source code isn't good enough. You'll have to investigate the symbol linkages or step through at runtime.

**Comment by peterpans01** [comment]
Yes, I understand that, so I use some code to check that (I put the file AgeRegression at HOME folder to check): 
```
#include <QCoreApplication>
#include <QByteArray>
#include <QFile>
#include <QDebug>

int main(int argc, char *argv[])
{
    QCoreApplication a(argc, argv);
    QFile f("/user/XXXXXX/home/AgeRegression");
      if (f.exists())
          qDebug() << "File exists";
      else
          qDebug() << "Missing file";
    QByteArray qb = f.readAll();
    qb = qUncompress(qb);
    qDebug()<<"Successfully";
    const char *qt_version = qVersion();
    qDebug()<< QString(qt_version);
    return a.exec();
}
```

And the result is:
File exists
QIODevice::read: device not open
qUncompress: Input data is corrupted
Successfully
"5.3.2"

```

**Comment by peterpans01** [comment]
from the documentation of Qt:
Note: If you want to use this function to uncompress external data that was compressed using zlib, you first need to prepend a four byte header to the byte array containing the data. The header must contain the expected length (in bytes) of the uncompressed data, expressed as an unsigned, big-endian, 32-bit integer.

**Comment by jklontz** [comment]
Thanks for the clarification, regrettably I can't reproduce the issue :/

**Comment by peterpans01** [comment]
There are some way you can provide the length of uncompressed data?
